packages/toggle-core/src/Read/ChainFeatureFinder.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pheature\Core\Toggle\Read;
+
+use Pheature\Core\Toggle\Exception\FeatureNotFoundException;
+use Pheature\Core\Toggle\Exception\FeatureNotFoundInChainException;
+
+class ChainFeatureFinder implements FeatureFinder
+{
+    /** @var FeatureFinder[] */
+    private array $featureFinders;
+
+    public function __construct(FeatureFinder ...$featureFinders)
+    {
+        $this->featureFinders = $featureFinders;
+    }
+
+    public function get(string $featureId): Feature
+    {
+        foreach ($this->featureFinders as $featureFinder) {
+            try {
+                return $featureFinder->get($featureId);
+            } catch (FeatureNotFoundException $exception) {
```

Comment:
If we don't catch this exception we are filling the contract anyway, because the factory should throw the expected exception type.

What we can do to specify that the exception was thrown at the Chain factory is to create another named constructor on the same `FeatureNotFoundException` class, this way we don't need to create another custom exception type.

What do you think?

Reply:
Great catch!! Totally agree with you, and maybe we could use the same named constructor because in fact, the `Chain FeatureFinder` is a `FeatureFinder`. What do you think?

Reply:
@kpicaza , I've just noticed that we cannot throw an interface 😅 , so we need to implement a custom exception like `packages/toggle-crud-psr11-factories/vendor/pheature/inmemory-toggle/src/InMemoryFeatureNotFound.php` 😕

Reply:
Sorry @xserrat, you are 100% right, I got confused with strategy and segment factories exceptions, we can avoid creating a new instance. :pray::pray::pray:

Reply:
Don't worry @kpicaza , I was confused too 😂 , problem solved!